python/cudf/cudf/core/column/decimal.py

```diff
@@ -98,7 +152,11 @@ def binary_operator(self, op, other, reflect=False):
         elif op in ("eq", "ne", "lt", "gt", "le", "ge"):
             if not isinstance(
                 other,
-                (DecimalColumn, cudf.core.column.NumericalColumn, cudf.Scalar),
+                (
+                    Decimal64Dtype,
```

Comment:
```suggestion
                    Decimal64Column,
```

I think this is why some of the binaryop tests are failing.